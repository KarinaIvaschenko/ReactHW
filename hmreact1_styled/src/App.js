import React from "react";
import Modal from "./components/Modal/Modal";
import "./app.scss";
import Button from "./components/Button/Button";
import { StyledButtons } from "./styledApp";
export default class App extends React.Component {
    state = { modal1: false, modal2: false };
    handlerOpenModal = (modalID) => {
        this.setState({ [modalID]: !this.state[modalID] });
    };

    render() {
        return (
            <>
                <StyledButtons>
                    <Button
                        onClick={(event) => this.handlerOpenModal("modal1")}
                        backgroundColor="#b3382c"
                        text="Open first modal"
                        isDisabled={this.state.modal2 ? true : false}
                    />
                    <Button
                        onClick={(event) => this.handlerOpenModal("modal2")}
                        backgroundColor="#f0dd2f"
                        text=" Open second modal"
                        isDisabled={this.state.modal1 ? true : false}
                    />
                </StyledButtons>
                {this.state.modal1 && (
                    <Modal
                        header={"Header"}
                        closeButton
                        $bgWind={true}
                        id="modal1"
                        close={this.handlerOpenModal}
                        text={"Text"}
                        actions={
                            <>
                                <Button
                                    onClick={(event) =>
                                        this.handlerOpenModal("modal1")
                                    }
                                    width="100px"
                                    backgroundColor="#b3382c"
                                    text="Ok"
                                />
                                <Button
                                    onClick={(event) =>
                                        this.handlerOpenModal("modal1")
                                    }
                                    width="100px"
                                    backgroundColor="#b3382c"
                                    text="Cancel"
                                />
                            </>
                        }
                    />
                )}
                {this.state.modal2 && (
                    <Modal
                        header={"Header"}
                        closeButton
                        id="modal2"
                        backgroundColor="#f0dd2f"
                        close={this.handlerOpenModal}
                        text={"Text"}
                        actions={
                            <>
                                <Button
                                    onClick={(event) =>
                                        this.handlerOpenModal("modal2")
                                    }
                                    backgroundColor="#f0dd2f"
                                    text={"Ok"}
                                />
                                <Button
                                    onClick={(event) =>
                                        this.handlerOpenModal("modal2")
                                    }
                                    backgroundColor="#f0dd2f"
                                    text={"Cancel"}
                                />
                            </>
                        }
                    />
                )}
            </>
        );
    }
}
