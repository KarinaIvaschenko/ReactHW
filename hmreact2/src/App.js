import React from "react";
import ProductList from "./components/Product/ProductList";
import Header from "./components/Header";
import Footer from "./components/Footer";
import Modal from "./components/Modal/Modal";
import Button from "./components/Button";
import "./app.scss";
import sendRequest from "./helpers/sendRequest";

export default class App extends React.Component {
    state = {
        favorites: JSON.parse(localStorage.getItem("favorites")) || [],
        cart: JSON.parse(localStorage.getItem("cart")) || [],
        modal: false,
        readyToCart: null,
        products: [],
    };

    componentDidMount() {
        sendRequest("product.json").then((data) => {
            this.setState({ products: data });
        });
    }
    componentDidUpdate() {
        if (this.state.favorites.length) {
            localStorage.setItem(
                "favorites",
                JSON.stringify(this.state.favorites)
            );
        }
        if (this.state.cart.length) {
            localStorage.setItem("cart", JSON.stringify(this.state.cart));
        }
    }
    addFavorites = (card) => {
        const isFavorites = this.state.favorites.some(
            (item) => item.id === card.id
        );
        if (!isFavorites) {
            this.setState((prev) => ({
                favorites: [...prev.favorites, card],
            }));
        }
    };
    removeFavorites = (card) => {
        const newFavorites = this.state.favorites.filter(
            (item) => item.id !== card.id
        );
        this.setState(() => ({
            favorites: newFavorites,
        }));
    };

    addToCart = (card) => {
        const isInCart = this.state.cart.some((item) => item.id === card.id);
        if (!isInCart) {
            this.setState((prev) => ({
                cart: [...prev.cart, card],
                modal: !prev.modal,
                readyToCart: null,
            }));
        }
    };

    addReadyToCart = (card) => {
        this.setState(() => ({
            readyToCart: card,
        }));
    };

    toggleModal = () => {
        this.setState((prev) => ({
            modal: !prev.modal,
        }));
    };

    render() {
        const { favorites, cart } = this.state;
        return (
            <>
                <Header countStar={favorites.length} countCart={cart.length} />
                <main className="main">
                    <ProductList
                        addFavorites={this.addFavorites}
                        removeFavorites={this.removeFavorites}
                        openModal={this.toggleModal}
                        addToCart={this.addToCart}
                        addReadyToCart={this.addReadyToCart}
                        products={this.state.products}
                    />
                    {this.state.modal && (
                        <Modal
                            text="Вы точно хотите добавить в корзину?"
                            header="Подтвердение"
                            close={this.toggleModal}
                            actions={
                                <>
                                    <Button
                                        onClick={() => {
                                            this.addToCart(
                                                this.state.readyToCart
                                            );
                                        }}
                                        text={"Добавить"}
                                        backgroundColor="#d3c1d9"
                                    />
                                    <Button
                                        onClick={this.toggleModal}
                                        backgroundColor="#d3c1d9"
                                        text={"Отмена"}
                                    />
                                </>
                            }
                        />
                    )}
                </main>
                <Footer />
            </>
        );
    }
}
