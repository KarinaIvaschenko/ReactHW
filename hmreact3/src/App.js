import { useEffect, useState } from "react";
import { Outlet } from "react-router-dom";
import AppRoutes from "./routes";
import Header from "./components/Header";
import "./app.scss";
import Footer from "./components/Footer/Footer";
import sendRequest from "./helpers/sendRequest";
const App = () => {
    const [favorites, setFavorites] = useState(
        JSON.parse(localStorage.getItem("favorites")) || []
    );
    const [cart, setCart] = useState(
        JSON.parse(localStorage.getItem("cart")) || []
    );
    const [products, setProducts] = useState([]);
    const [readyToCart, setReadyToCart] = useState(null);

    useEffect(() => {
        sendRequest("product.json").then((data) => {
            setProducts(data);
        });
    }, []);

    useEffect(() => {
        if (favorites.length) {
            localStorage.setItem("favorites", JSON.stringify(favorites));
        }
        if (cart.length) {
            localStorage.setItem("cart", JSON.stringify(cart));
        }
    }, [favorites, cart]);

    const addReadyToCart = (card) => {
        setReadyToCart(card);
    };

    const addToCart = (card) => {
        const isInCart = cart.some((item) => item.id === card.id);
        if (!isInCart) {
            setCart((prev) => [...prev, card]);
            setReadyToCart(null);
        }
    };

    const removeCart = (card) => {
        const newCart = cart.filter((item) => item.id !== card.id);
        setCart(() => newCart);
        localStorage.setItem("cart", JSON.stringify(newCart));
    };

    const addFavorites = (card) => {
        const isFavorites = favorites.some((item) => item.id === card.id);
        if (!isFavorites) {
            setFavorites((prev) => [...prev, card]);
        }
    };
    const removeFavorites = (card) => {
        const newFavorites = favorites.filter((item) => item.id !== card.id);
        setFavorites(() => newFavorites);
        localStorage.setItem("favorites", JSON.stringify(newFavorites));
    };

    return (
        <>
            <Header countStar={favorites.length} countCart={cart.length} />
            <main className="main">
                <AppRoutes
                    favorites={favorites}
                    cart={cart}
                    products={products}
                    setFavorites={setFavorites}
                    setCart={setCart}
                    addFavorites={addFavorites}
                    removeFavorites={removeFavorites}
                    readyToCart={readyToCart}
                    addToCart={addToCart}
                    removeCart={removeCart}
                    addReadyToCart={addReadyToCart}
                />
                <Outlet />
            </main>
            <Footer />
        </>
    );
};

export default App;
