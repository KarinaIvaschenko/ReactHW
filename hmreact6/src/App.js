import { useEffect } from "react";
import { Outlet } from "react-router-dom";
import AppRoutes from "./routes";
import Header from "./components/Header";
import "./app.scss";
import Footer from "./components/Footer/Footer";
import { useDispatch, useSelector } from "react-redux";
import { actionsFetchProducts } from "./store/cards/actionsCards";

const App = () => {
    const favorites = useSelector((state) => state.favorites.favorites);
    const cart = useSelector((state) => state.cart.cart);

    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(actionsFetchProducts());
    }, []);

    useEffect(() => {
        if (favorites.length) {
            localStorage.setItem("favorites", JSON.stringify(favorites));
        }
        if (cart.length) {
            localStorage.setItem("cart", JSON.stringify(cart));
        }
    }, [favorites, cart]);

    return (
        <>
            <Header countStar={favorites.length} countCart={cart.length} />
            <main className="main">
                <AppRoutes />
                <Outlet />
            </main>
            <Footer />
        </>
    );
};

export default App;
