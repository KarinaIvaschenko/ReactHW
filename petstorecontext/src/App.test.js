import "@testing-library/jest-dom";
import App from "./App";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "./store/reducer";
import { MemoryRouter } from "react-router-dom";

describe("App", () => {
    const store = configureStore({ reducer: rootReducer });
    test("snapshot app", () => {
        const { container } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <App />
                </MemoryRouter>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
