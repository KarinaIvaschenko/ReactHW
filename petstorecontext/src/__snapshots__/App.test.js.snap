// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot app 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header__option"
    >
      <nav
        class="navigation"
      >
        <ul
          class="navigation__list"
        >
          <li
            class="navigation__item"
          >
            <a
              aria-current="page"
              class="navigation__link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="/favorites"
            >
              Favorites
            </a>
          </li>
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="/cart"
            >
              Cart
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="icon"
      >
        <a
          data-testid="favorites"
          href="/favorites"
        >
          <span
            class="chosenCount"
            data-testid="countStar"
          >
            0
            <svg>
              chosenSvg.svg
            </svg>
          </span>
        </a>
        <a
          data-testid="cart"
          href="/cart"
        >
          <span
            class="cartCount"
            data-testid="countCart"
          >
            0
            <img
              alt=""
              class="cartImg"
              src="shopingcart.ico"
            />
          </span>
        </a>
      </div>
    </div>
  </header>
  <main
    class="main"
  >
    <div
      class="wrapper"
    />
  </main>
  <footer
    class="footer"
  />
</div>
`;
